app/services/export/s3_uploader.rb

```diff
@@ -25,17 +26,23 @@ def upload
       )
       raise "Unexpected response." unless response&.etag
 
-      OpenStruct.new(
-        url: bucket.object(filename).public_url,
-        timestamped_filename: filename
-      )
+      if public
+        OpenStruct.new(
+          url: bucket.object(filename).public_url,
+          timestamped_filename: filename
+        )
+      else
+        OpenStruct.new(
+          timestamped_filename: filename
+        )
+      end
```

Comment:
I think I prefer treating the two return OpenStructs as kind of separate/distinct objects here rather than considering them the same object with different fields, if that makes sense at all?

We'll be removing this conditional behaviour pretty soon anyway, so it shouldn't be a concern in the codebase for too much longer.